### PR TITLE
Modal and Form for login. Also added react-hook-form.

### PR DESCRIPTION
**Pull request by LukavSlambert**

### CODE REVIEW

File: src/components/forms/LoginForm.jsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    return (
+        <form onSubmit={handleSubmit(onSubmit)}>
+            <input {...register('email', {
+                required: "Email is required",
+            })}
+                type='email'
+                placeholder='Email'
+            />
+            {errors.email && (<p>{`${errors.email.message}`}</p>)}
```
Comment (LimKab) [inline src/components/forms/LoginForm.jsx:21]: I think you don't need the `${}` here

File: src/components/forms/LoginForm.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+
+import { useForm } from 'react-hook-form';
+
+export default function LoginForm({ onSubmit }) {
```
Comment (LimKab) [inline src/components/forms/LoginForm.jsx:4]: Is the login going to have a password confirmation or is it the signup? or is it like reusable for both?

File: src/components/forms/LoginForm.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            <input {...register('confirmPassword', {
+                required: "Confirm password is required",
+                validate: (value) =>
+                    value === getValues("password") || "Passwords must match"
+            })}
+                type="password"
+                placeholder='Confirm password'
+            />
+            {errors.confirmPassword && (<p>{`${errors.confirmPassword.message}`}</p>)}
+            <input type="submit" />
```
Comment (LimKab) [inline src/components/forms/LoginForm.jsx:42]: submit should be a button, I understand if you prefer to leave it to the fe team (:

File: src/modals/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    transform: "translate(-50%, -50%)",
+    width: 400,
+    bgcolor: "background.paper",
+    border: "2px solid #000",
+    boxShadow: 24,
+    p: 4,
+};
+
+export default function Login() {
+    const [open, setOpen] = React.useState(false);
```
Comment (LimKab) [inline src/modals/Login.jsx:21]: do you usually write useState like this or is it from MUI thing?

File: src/modals/Login.jsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    const handleClose = () => setOpen(false);
+
+    const onSubmit = async (data) => {
+        handleClose()
```
Comment (LimKab) [inline src/modals/Login.jsx:25]: I guess here is for us to continue with the post.. do have a route for it?

File: src/modals/Login.jsx (hunk 4 above)
Comment (LukavSlambert) [reply src/modals/Login.jsx:21]: MUI thing

File: src/modals/Login.jsx (hunk 5 above)
Comment (LukavSlambert) [reply src/modals/Login.jsx:25]: Yes, evreything ready from BE. I going to do authentication now, if you dont mind

File: src/components/forms/LoginForm.jsx (hunk 2 above)
Comment (LukavSlambert) [reply src/components/forms/LoginForm.jsx:4]: I suggest to do reusable form for both

File: src/components/forms/LoginForm.jsx (hunk 1 above)
Comment (LukavSlambert) [reply src/components/forms/LoginForm.jsx:21]: Doesn't work without it.

File: src/components/forms/LoginForm.jsx (hunk 3 above)
Comment (LukavSlambert) [reply src/components/forms/LoginForm.jsx:42]: Honestly didn't noticed this :) I followed tutorial for react-hook-form :)

**Review by LimKab — APPROVED** [review]

**Pull request merged**